assets/src/dashboard/app/views/editorSettings/index.js

```diff
@@ -21,53 +21,141 @@ import { __ } from '@wordpress/i18n';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
 
 /**
  * Internal dependencies
  */
+import { Layout } from '../../../components';
 import { ApiContext } from '../../api/apiProvider';
+import { useConfig } from '../../config';
+import { PageHeading } from '../shared';
 import GoogleAnalyticsSettings from './googleAnalytics';
-import { Wrapper, Header, Heading, Main } from './components';
+import { Wrapper, Main } from './components';
+import PublisherLogoSettings from './publisherLogo';
 
 function EditorSettings() {
   const {
     actions: {
       settingsApi: { fetchSettings, updateSettings },
+      mediaApi: { uploadMedia, fetchMediaById },
     },
     state: {
-      settings: { googleAnalyticsId },
+      settings: { activePublisherLogoId, googleAnalyticsId, publisherLogoIds },
+      media: { isLoading: isMediaLoading, uploadedMediaIds, publisherLogos },
     },
   } = useContext(ApiContext);
 
+  const { capabilities: { canUploadFiles } = {}, maxUpload } = useConfig();
+
+  const [mediaError, setMediaError] = useState('');
+  /**
+   * WP settings references publisher logos by ID.
+   * We must retrieve the media for those ids from /media when present
+   * Summarized below:
+   * FETCH - fetch settings to know what IDs to load media for
+   * then fetch media when publisherLogoIds are in state with useEffect
+   * ADD - first upload new logo to media. use the new ID(s) to update settings.
+   * REMOVE - send id to remove to settings. Not deleting media, just removing from logos.
+   */
+
   useEffect(() => {
     fetchSettings();
   }, [fetchSettings]);
 
-  const handleCompleteUpdateId = useCallback(
-    ({ newGoogleAnalyticsId }) => {
-      const updatedSettings = {
-        googleAnalyticsId:
-          typeof newGoogleAnalyticsId === 'string' && newGoogleAnalyticsId,
-      };
+  useEffect(() => {
+    if (uploadedMediaIds.length > 0) {
+      updateSettings({ publisherLogoIds: uploadedMediaIds });
+    }
+  }, [updateSettings, uploadedMediaIds]);
 
-      updateSettings(updatedSettings);
+  useEffect(() => {
+    if (publisherLogoIds.length > 0) {
+      fetchMediaById(publisherLogoIds);
+    }
+  }, [fetchMediaById, publisherLogoIds]);
+
+  const handleUpdateGoogleAnalyticsId = useCallback(
+    (newGoogleAnalyticsId) =>
+      updateSettings({ googleAnalyticsId: newGoogleAnalyticsId }),
+    [updateSettings]
+  );
+
+  const handleAddLogos = useCallback(
+    (files) => {
+      const isFileSizeWithinMaxUpload = files.every(
+        (file) => file.size <= maxUpload
+      );
+
+      if (!isFileSizeWithinMaxUpload) {
+        const errorText =
+          files.length > 1
+            ? __(
+                'Sorry, this file is too big. Make sure your logo is under 100 MB.',
+                'web-stories'
+              )
+            : __(
+                'Sorry, one or more of these files are too big. Make sure your logos are all under 100 MB.',
```

Comment:
Where is this 100 MB limit coming from?

For these kind of values it's better to use `sprintf` with placeholders. So. `[...] under %s.`

Reply:
ah thank you. I was using 100 MB because that's what's coming through as the maxUpload in config and what the other upload max message says natively in WordPress. That said, bit off too much in one PR and let this slide. Thanks for calling it out. Added in a util to convert the bytes received as maxUpload into a readable number - (`utils/formatBytes.js`) and updated translations to use sprintf. 

If you know a more graceful way to handle the plural/singular here I'm all ears.